src/header.tsx

```diff
@@ -62,13 +66,17 @@ export const FilesCardHeader = ({ currentFilter, onFilterChange, isGrid, setIsGr
     );
 };
 
-const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
-    const [isOpen, setIsOpen] = useState(false);
-    const onToggleClick = isOpen => setIsOpen(!isOpen);
-    const onSelect = (ev, itemId) => {
-        setIsOpen(false);
-        setSortBy(itemId);
-        localStorage.setItem("files:sort", itemId);
+const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }:
+    { isGrid: boolean, setIsGrid: React.Dispatch<React.SetStateAction<boolean>>,
+      sortBy: string, setSortBy: React.Dispatch<React.SetStateAction<string>>}) => {
+    const [isOpen, setIsOpen] = useState<boolean>(false);
+    const onToggleClick = (isOpen: boolean) => setIsOpen(!isOpen);
+    const onSelect = (_ev?: React.MouseEvent, itemId?: string | number) => {
+        if (itemId && typeof itemId === "string") {
```

Comment:
So `console.assert`?

Reply:
Concluding, `console.assert` logs, `assert` is nodejs. I am adding an `else` to with a `console.error`, we should make an issue @ PatternFly about this likely this is typed wrong?

Reply:
I think having an itemId as a number is definitely possible in case you wanted to do that (although maybe they'd be able to improve the situation there through the use of generics).  I'm not sure about it, but the `undefined` thing feels a bit like someone didn't understand contravariance when they wrote those type definitions... like, the `?:` they added might have been meant to make it easier to implement the function for users ("ahh.. they can leave this argument out if they want...") but it actually makes it harder for us because we have to *also* accept both possibilities now...